From: Zhong Jianxin <azuwis@gmail.com>
Date: Fri, 24 Nov 2017 10:32:23 +0800
Subject: Restore special files encoding hack

---
 lib/libebook/ebook_chm.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/libebook/ebook_chm.cpp b/lib/libebook/ebook_chm.cpp
index 67d51e9..ece1d73 100644
--- a/lib/libebook/ebook_chm.cpp
+++ b/lib/libebook/ebook_chm.cpp
@@ -227,7 +227,7 @@ bool EBook_CHM::getTextContent( QString& str, const QString& url, bool internal_
 			buf.resize( length + 1 );
 			buf [length] = '\0';
 
-			str = internal_encoding ? (QString)( buf.constData() ) :  encodeWithCurrentCodec( buf.constData() );
+			str = internal_encoding ? encodeInternalWithCurrentCodec( buf.constData() ) :  encodeWithCurrentCodec( buf.constData() );
 			return true;
 		}
 	}
@@ -379,7 +379,7 @@ bool EBook_CHM::parseFileAndFillArray( const QString& file, QList< ParsedEntry >
 	QString src;
 	const int MAX_NEST_DEPTH = 256;
 
-	if ( !getTextContent( src, file ) || src.isEmpty() )
+	if ( !getTextContent( src, file, true ) || src.isEmpty() )
 		return false;
 
 /*
@@ -976,7 +976,7 @@ bool EBook_CHM::RecurseLoadBTOC( const QByteArray& tocidx,
 					return false;
 				}
 
-				name = encodeWithCurrentCodec( strings.data() + index);
+				name = encodeInternalWithCurrentCodec( strings.data() + index);
 			}
 			else
 			{
@@ -997,7 +997,7 @@ bool EBook_CHM::RecurseLoadBTOC( const QByteArray& tocidx,
 				if ( tocoffset < 0 )
 					name.clear();
 				else
-					name = encodeWithCurrentCodec( strings.data() + tocoffset );
+					name = encodeInternalWithCurrentCodec( strings.data() + tocoffset );
 
 				// #URLTBL index
 				tocoffset = (int) UINT32ARRAY( topics.data() + (index * 16) + 8 );
@@ -1016,7 +1016,7 @@ bool EBook_CHM::RecurseLoadBTOC( const QByteArray& tocidx,
 					return false;
 				}
 
-				value = encodeWithCurrentCodec( urlstr.data() + tocoffset + 8 );
+				value = encodeInternalWithCurrentCodec( urlstr.data() + tocoffset + 8 );
 			}
 
 			EBookTocEntry entry;
