From: Zhong Jianxin <azuwis@gmail.com>
Date: Fri, 24 Nov 2017 15:50:57 +0800
Subject: Fix topics/index/home path encoding

---
 lib/libebook/ebook_chm.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/libebook/ebook_chm.cpp b/lib/libebook/ebook_chm.cpp
index 86f61af..a229ec8 100644
--- a/lib/libebook/ebook_chm.cpp
+++ b/lib/libebook/ebook_chm.cpp
@@ -85,7 +85,7 @@ QString EBook_CHM::title() const
 
 QUrl EBook_CHM::homeUrl() const
 {
-	return pathToUrl( m_home );
+	return pathToUrl( encodeWithCurrentCodec(m_home) );
 }
 
 bool EBook_CHM::hasFeature(EBook::Feature code) const
@@ -113,7 +113,7 @@ bool EBook_CHM::getTableOfContents( QList<EBookTocEntry> &toc ) const
 	// Parse the plain text TOC
 	QList< ParsedEntry > parsed;
 
-	if ( !parseFileAndFillArray( m_topicsFile, parsed, false ) )
+	if ( !parseFileAndFillArray( encodeWithCurrentCodec(m_topicsFile), parsed, false ) )
 		return false;
 
 	// Find out the root offset, and reduce the indent level to it
@@ -146,7 +146,7 @@ bool EBook_CHM::getIndex(QList<EBookIndexEntry> &index) const
 	// Parse the plain text index
 	QList< ParsedEntry > parsed;
 
-	if ( !parseFileAndFillArray( m_indexFile, parsed, true ) )
+	if ( !parseFileAndFillArray( encodeWithCurrentCodec(m_indexFile), parsed, true ) )
 		return false;
 
 	// Find out the root offset, and reduce the indent level to it
